toast-tk-runtime/src/main/java/io/toast/tk/runtime/AbstractScenarioRunner.java

```diff
@@ -17,141 +13,133 @@
+			final String... scenarios
+	) throws Exception {
+		runScenario(scenarios);
+	}
+
+	private void runScenario(
+			final String... scenarios
+	) throws Exception {
+		final List<ITestPage> testPages = new ArrayList<>();
+		initEnvironment();
+		for (final String fileName : scenarios) {
+			LOG.info("Start main test parser: {}", fileName);
+
+			List<String> lines = FileHelper.getScript(fileName);
+			final ITestPage result = runTestPage(new TestParser().parse(lines, fileName));
+			testPages.add(result);
 		}
 
-    	final URL resource = this.getClass().getClassLoader().getResource(fileName);
-        if (resource == null) {
-            throw new FileNotFoundException(fileName);
-        }
-        return new File(resource.getFile());
-    }
-
-    private ITestPage runScript(
-    	final File file,
-    	final String script
-    ) throws IOException {
-    	final TestParser testParser = new TestParser();
-        ITestPage result = file == null ? testParser.readString(script, null) : testParser.parse(file.getPath());
-        final TestRunner runner = injector.getInstance(TestRunner.class);
-        if(this.presetRepoFromWebApp) {
-        	final String repoWiki = RestUtils.downloadRepositoryAsWiki();
-        	final TestParser parser = new TestParser();
-        	final ITestPage repoAsTestPageForConvenience = parser.readString(repoWiki, null);
-            runner.run(repoAsTestPageForConvenience);
-        } 
-        else if (this.localRepositoryTestPage != null) {
-            runner.run(this.localRepositoryTestPage);
-        }
-        beginTest();
-        result = runner.run(result);
-        createAndOpenReport(result);
-        endTest();
-        return result;
-    }
-
-    protected void createAndOpenReport(
-    	final ITestPage testPage
-    ) {
-    	final String generatePageHtml = htmlReportGenerator.generatePageHtml(testPage);
-    	final String path = getReportsFolderPath();
-        final String pageName = testPage.getName();
-        htmlReportGenerator.writeFile(generatePageHtml, pageName, path);
-        openReport(path, pageName);
-    }
+		tearDownEnvironment();
+
+		LOG.info("{}file(s) processed", scenarios.length);
+		RunUtils.printResult(testPages);
+	}
+
+	public final void runRemote(
+			final String... scenarios
+	) throws Exception {
+		this.presetRepoFromWebApp = true;
+		run(scenarios);
+	}
+
+	public final void runRemoteScript(
+			final String script
+	) throws Exception {
+		this.presetRepoFromWebApp = true;
+		runScript(script);
+	}
+
+	public void runLocalScript(
+			final String wikiScenario,
+			final String repoWiki,
+			final IReportUpdateCallBack callback
+	) throws IOException {
+		this.progressReporter.setReportCallBack(callback);
+		final TestParser parser = new TestParser();
+		this.localRepositoryTestPage = parser.readString(repoWiki, null);
+		runScript(wikiScenario);
+	}
+
+	private InputStream readTestFile(final String fileName) throws IOException, URISyntaxException {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Private method 'readTestFile' is never used. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.talanlabs.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)